File: Ladder 11/Python/042.Magic Numbers.py
x = input()
while len(x)>0:
    # print(x[:3] , x[3:])
    if x[:3]=="144":
        x = x[3:]
        # print(x)
    elif x[:2]=="14":
        # print("2")
        x = x[2:]
    elif x[:1]=="1":
        # print("3")
        x = x[1:]
    else:
        print("NO")
        break
else:
    print("YES")

File: Ladder 11/042 Magic Numbers.cpp
#include <iostream>
#include <string>
using namespace std;
int main() {
    string a;
    cin>>a;
    int b=a.size();
    int c=0;
    for (int i=0;i<b;i++)
    {
        if (a[i]=='1' && a[i+1]=='4' && a[i+2]=='4')
        i=i+2;
        else if (a[i]=='1' && a[i+1]=='4')
        i=i+1;
        else if (a[i]=='1')
        i=i;
        else 
        c=1;
    }
    if (c==0)
    cout<<"YES";
    else 
    cout<<"NO";
    return 0;
}